python/cudf/cudf/core/column/column.py

```diff
@@ -968,7 +967,8 @@ def astype(self, dtype: Dtype, copy: bool = False) -> ColumnBase:
             col = self
         if self.dtype == dtype:
             return col
-        if _is_categorical_dtype(dtype):
+        dtype = cudf.dtype(dtype)
+        if isinstance(dtype, cudf.CategoricalDtype):
```

Comment:
I think replacing `_is_categorical_dtype` calls else where with `isinstance` checks is the right thing but places like these where users can pass `.astype('category')`, we will need `_is_categroical_dtype` because doing a `cudf.dtype('category')` will lead to different behaviors:
```
In [1]: import cudf

In [2]: cudf.dtype("category")
Out[2]: CategoricalDtype(categories=None, ordered=None, categories_dtype=None)
```